packages/fxa-settings/src/components/DropDownAvatarMenu/index.tsx

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import Avatar from '../Avatar';
+import AlertBar from '../AlertBar';
+import { useAccount } from '../../models';
+import sentryMetrics from 'fxa-shared/lib/sentry';
+import { clearSignedInAccountUid } from '../../lib/cache';
+import { useClickOutsideEffect, useBooleanState } from 'fxa-react/lib/hooks';
+import { useEscKeydownEffect } from '../../lib/hooks';
+import { ReactComponent as SignOut } from './sign-out.svg';
+
+export const DESTROY_SESSION_MUTATION = gql`
+  mutation destroySession($input: DestroySessionInput!) {
+    destroySession(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const DropDownAvatarMenu = () => {
+  const { displayName, primaryEmail } = useAccount();
+  const [isRevealed, setRevealed] = useState(false);
+  const toggleRevealed = () => setRevealed(!isRevealed);
+  const avatarMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(
+    setRevealed
+  );
+  useEscKeydownEffect(setRevealed);
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const dropDownId = 'drop-down-avatar-menu';
+
+  // TODO: FXA-6249
```

Comment:
FXA-6249 doesn't seem to exist, but I assume it's that one about DRYing up the useMutation. If #6292 is merged before this, you'll be able to use `useHandledMutation` properly here.

Reply:
Oh good catch, yeah I meant #6249 which is FXA-2450.

I think we'll still have a TODO here since that ticket also deals with figuring out what should report to Sentry, what error text copy to grab from Meridel, and cleaning up/standardizing all of the mutations and properly styling the alertbar accordingly.